### PR TITLE
[TearingEngine] Fix compilation and warnings with SOFA master

### PR DESCRIPTION
**Pull request by hugtalbot**
Fix and clean done while applying the lifecycle changes in SOFA for v24.06

### CODE REVIEW

**Review by epernod — APPROVED** [review]
ok for me, double check with @rmolazem

File: src/Tearing/TearingEngine.inl [hunk 1]
```diff
@@ -484,6 +485,7 @@ inline bool TearingEngine<DataTypes>::computeEndPointsNeighboringTriangles(Coord
 template<class DataTypes>
 inline bool TearingEngine<DataTypes>::computeIntersectionNeighborTriangle(Coord normalizedFractureDirection,Coord Pa ,Coord& Pb, Real& t)
 {
+    SOFA_UNUSED(Pa);
```
Comment (rmolazem) [inline src/Tearing/TearingEngine.inl:488]: The 'Pa' parameter can be removed from this function's argument. Consequently, it should also be removed from the function 'computeEndPointsNeighboringTriangles' accordingly.

File: Tearing_test/TearingEngine_test.cpp [hunk 2]
```diff
@@ -70,7 +70,7 @@ class TearingEngine_test : public BaseSimulationTest
     void TearDown() override
     {
         if (m_instance.root != nullptr)
-            sofa::simulation::getSimulation()->unload(m_instance.root);
+            sofa::simulation::node::unload(m_instance.root);
```
Comment (rmolazem) [inline Tearing_test/TearingEngine_test.cpp:73]: I do not know what is this change about.

**Review by rmolazem — CHANGES_REQUESTED** [review]

File: src/Tearing/TearingEngine.inl (hunk 1 above)
Comment (epernod) [reply src/Tearing/TearingEngine.inl:488]: ok, good to know, we will remove it in another PR

File: Tearing_test/TearingEngine_test.cpp (hunk 2 above)
Comment (epernod) [reply Tearing_test/TearingEngine_test.cpp:73]: This is the propagation from change done in the SOFA core code.

**Pull request merged**